File: test_scripts/python/client_test_autogen.py
# from runagent import RunAgentClient

# ra = RunAgentClient(
#     agent_id="20cf0046-15f6-4b9e-84b4-9be60c031fdc",
#     entrypoint_tag="autogen_invoke",
#     local=True
# )

# agent_result = ra.run(
#     task="What is Autogen??"
# )
# print(agent_result)
# ==================================

# from runagent import RunAgentClient

# ra = RunAgentClient(
#     agent_id="20cf0046-15f6-4b9e-84b4-9be60c031fdc",
#     entrypoint_tag="autogen_step_stream",
#     local=True
#     )

# for chunk in ra.run(
#     task="Analyze the benefits of remote work for software teams"
# ):
#     if chunk["type"] == "TextMessage":
#         print(f"\n\n{chunk['source']}:")
#         print("=====================")
#         print(chunk["message"])

# ==================================

# from runagent import RunAgentClient

# ra = RunAgentClient(
#     agent_id="20cf0046-15f6-4b9e-84b4-9be60c031fdc",
#     entrypoint_tag="autogen_token_stream",
#     local=True
#     )

# for chunk in ra.run(
#     task="Analyze the benefits of remote work for software teams"
# ):  
#     if chunk["type"] == "ModelClientStreamingChunkEvent":
#         print(chunk["delta"], end="")

File: test_scripts/go/client_test_autogen.go
package main

import (
	"context"
	"fmt"
	"log"
	"time"

	"github.com/runagent-dev/runagent-go/pkg/client"
)

func main() {
	// Create client
	agentClient, err := client.New(
		"<agent_id>",     // agentID
		"autogen_invoke", // entrypointTag
		true,             // local
	)
	if err != nil {
		log.Fatalf("Failed to create client: %v", err)
	}
	defer agentClient.Close()

	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
	defer cancel()

	// Run the agent
	result, err := agentClient.Run(ctx, map[string]interface{}{
		"task": "what is autogen",
	})
	if err != nil {
		log.Fatalf("Failed to run agent: %v", err)
	}

	fmt.Printf("Result: %v\n", result)
}
